pcdscalc/diffraction.py

```diff
@@ -0,0 +1,151 @@
+"""
+Module that handles calculations for diffraction/crystallography.
+
+Assume all the calculations that expect energy as input will
+require energy in eV.
+"""
+import numpy as np
+
+from .constants import lattice_parameters, units, alias
+
+# wavelength/energy = 12398.4 (A) / E(eV)
+WAVELENGTH_TO_ENERGY_LAMBDA = 12398.4
+
+
+def energy_to_wavelength(energy):
+    """
+    Compute photon wavelength in m.
+
+    Parameters
+    ----------
+    energy : number
+        Photon energy in eV.
+
+    Returns
+    -------
+    wavelength : float
+        Wavelength [m].
+    """
+    return (WAVELENGTH_TO_ENERGY_LAMBDA / energy) / units["ang"]
+
+
+def wavelength_to_energy(wavelength):
+    """
+    Compute photon energy in eV.
+
+    Parameters
+    ----------
+    wavelength : number
+        The photon wavelength in m.
+
+    Returns
+    -------
+    energy : number
+        Photon Energy in eV.
+    """
+    return WAVELENGTH_TO_ENERGY_LAMBDA / (wavelength * units["ang"])
+
+
+def get_lom_geometry(energy, material_id, reflection):
+    """
+    Calculate the Large Offset Monochromator's crystal geometry.
+
+    Parameters
+    ----------
+    energy : number
+        Photon energy in eV.
+    material_id : str
+        Chemical formula. E.g.: `Si`.
+    reflection : tuple
+        The reflection. E.g.: `(1,1,1)`.
+
+    Returns
+    -------
+    thm, zm : tuple
+    """
+    th = np.radians(bragg_angle(material_id, reflection, energy))
+    zm = 300 / np.tan(2 * th)
+    thm = np.rad2deg(th)
+    return thm, zm
+
+
+def bragg_angle(material_id, hkl, energy):
+    """
+    Compute the Bragg angle in deg.
+
+    Computes the bragg angle of the specified material, reflection and photon
+    energy.
+
+    Parameters
+    ----------
+    material_id : str
+        Chemical formula. Defaults to `Si`.
+    hkl : tuple
+        The reflection indices. Defaults to `(1,1,1)`.
+    energy : number
+        The photon energy in eV.
+
+    Returns
+    -------
+    theta : number
+        Theta in degrees.
+    """
+    material_id = alias.get(material_id, material_id)
+    d = d_space(material_id, hkl)
+    theta = np.arcsin(energy_to_wavelength(energy) / 2 / d)
+    return np.degrees(theta)
+
+
+def d_space(material_id, hkl):
+    """
+    Compute the d spacing (m) of the specified material and reflection.
+
+    Parameters
+    ----------
+    material_id : str
+        Chemical fomula. E.g.: `Si`
+    hkl : tuple
+        Miller Indices, the reflection. E.g.: `(1,1,1)`
+
+    Returns
+    -------
+    d : number
+        Inverse d_space squared.
```

Comment:
I totally made this up - i have no idea what this is actually: `inv_d_sqr`

Reply:
this is the lattice plane spacing d in Bragg's Law (https://en.wikipedia.org/wiki/Bragg's_law), for the chosen plane indicated by the Miller indices.

Reply:
Thank you!